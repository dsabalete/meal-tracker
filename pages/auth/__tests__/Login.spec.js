import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import LoginView from '@/pages/auth/login.vue'

const build = (props) => {
  const wrapper = shallowMount(LoginView, {
    stubs: {
      NuxtLink: RouterLinkStub,
    },
    propsData: { ...props },
  })
  return wrapper
}

describe('Login View', () => {
  test('should be rendered', () => {
    const wrapper = build()
    expect(wrapper.exists()).toBeTruthy()
  })

  test('should match snapshot', () => {
    const wrapper = build()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should have a social login button', () => {
    const wrapper = build()
    const socialButton = wrapper.findComponent({ name: 'SocialLoginButton' })
    expect(socialButton.exists()).toBeTruthy()
  })
})
