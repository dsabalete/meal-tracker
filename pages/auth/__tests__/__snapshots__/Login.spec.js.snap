// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login View should match snapshot 1`] = `
<div
  class="h-full overflow-hidden"
>
  <div
    class="flex justify-end p-4"
  >
    <colorswitch-stub />
  </div>
   
  <h1
    class="text-center text-3xl font-bold"
  >
    Meal Tracker
  </h1>
   
  <div
    class="login-body mt-12 h-full flex flex-col justify-center items-center bg-light dark:bg-blue-main-800"
  >
    <socialloginbutton-stub
      text="Iniciar sesiÃ³n con google"
      type="google"
    />
     
    <p>
      iniciar sesion con google
    </p>
  </div>
</div>
`;
